hr_holidays_leave_auto_approve/models/hr_holidays.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# © 2016 ONESTEiN BV (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class HrHolidays(models.Model):
+    _inherit = "hr.holidays"
+
+    @api.model
+    @api.returns('self', lambda value: value.id)
+    def create(self, values):
+        result = super(HrHolidays, self).create(values)
+        if values.get('holiday_status_id'):
+            type_obj = self.env['hr.holidays.status']
+            type = type_obj.browse(values.get('holiday_status_id'))
```

Comment:
Please avoid using python's reserved names (`type` here) as variable name.

Also, you should use directly use `result.holiday_status_id` instead of `values.get('holiday_status_id`, which is the real data written on the record, because some methods called by the `super` could have changed the values.
Moreover, this will directly give you the recordset, without the need to call `browse`, and you can safely read fields on this without testing if it exists.

Finally, you can rewrite this part this way :
```
result = super(HrHolidays, self).create(values) 
if result.holiday_status_id.auto_approve:
    result.sudo().holiday_validate()
return result
```